src/backend/distributed/commands/role.c

```diff
@@ -0,0 +1,128 @@
+#include "postgres.h"
+
+#include "nodes/parsenodes.h"
+#include "utils/syscache.h"
+#include "access/table.h"
+#include "catalog/pg_authid.h"
+#include "utils/rel.h"
+#include "utils/builtins.h"
+#include "access/heapam.h"
+#include "distributed/worker_transaction.h"
+#include "distributed/commands/utility_hook.h"
+#include "distributed/master_protocol.h"
+
+List * ProcessAlterRoleStmt(AlterRoleStmt *stmt, const char *queryString);
+static const char * CreateSetPasswordQueryForWorkerNodes(const char *rolename);
+List * CreateSetPasswordQueriesOfUsersWithPassword(void);
+
+
+bool EnablePasswordPropagation = true;
+
+/*
+ * ProcessAlterRoleStmt actually creates the plan we need to execute for alter
+ * role statement.
+ */
+List *
+ProcessAlterRoleStmt(AlterRoleStmt *stmt, const char *queryString)
```

Comment:
I think this name is too broad, the `ALTER ROLE` could have lots of options, we're only considering `password` piece. So, we should probably rename it to `ProcessAlterRoleSetPasswordStmt()`